File: Consulta de Representación del conocimiento/Logica de predicados/Logica de predicados Java/src/logica/de/predicados/java/LogicaDePredicados.java
package logica.de.predicados.java;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author Daniel Aviles, DeltaTeam, DCCO-ESPE
 */
public class LogicaDePredicados {
    /*En este ejemplo de logica de predicados se
    tiene una clase perro con la funcion ladrar
    y con la funcion todosLosPerrosLadran se 
    busca verificar la premisa*/
    public static boolean todosLosPerrosLadran(List<Perro> perros) {
        for (Perro perro : perros) {
            if (!perro.ladrar()) {
                return false;
            }
        }
        return true;
    }
    
    public static void main(String[] args) {
        List<Perro> listaPerros = new ArrayList<>();
        listaPerros.add(new Perro("Max"));
        listaPerros.add(new Perro("Bella"));
        listaPerros.add(new Perro("Charlie"));
        
        if (todosLosPerrosLadran(listaPerros)) {
            System.out.println("Todos los perros ladran.");
        } else {
            System.out.println("No todos los perros ladran.");
        }
    }
    
}

File: Consulta de Representación del conocimiento/Logica de predicados/Logica de predicados.py
"""En este ejemplo de logica de predicados se
    tiene una clase perro con la funcion ladrar
    y con la funcion todosLosPerrosLadran se 
    busca verificar la premisa"""
class Perro:
    def __init__(self, nombre):
        self.nombre = nombre
    
    def ladrar(self):
        print(self.nombre + " está ladrando.")
        return True

def todosLosPerrosLadran(perros):
    for perro in perros:
        if not perro.ladrar():
            return False
    return True

listaPerros = [Perro("Max"), Perro("Bella"), Perro("Charlie")]

if todosLosPerrosLadran(listaPerros):
    print("Todos los perros ladran.")
else:
    print("No todos los perros ladran.")
